embedded/appendable/singleapp/single_app.go

```diff
@@ -362,18 +378,75 @@ func (aof *AppendableFile) Append(bs []byte) (off int64, n int, err error) {
 	bbLenBs := make([]byte, 4)
 	binary.BigEndian.PutUint32(bbLenBs, uint32(len(bb)))
 
-	n, err = aof.w.Write(bbLenBs)
+	n, err = aof.write(bbLenBs)
 	if err != nil {
-		return
+		return off, n, err
 	}
 
-	n, err = aof.w.Write(bb)
-	if err != nil {
-		return off, 4 + n, err
+	n, err = aof.write(bb)
+
+	return off, n + 4, err
+}
+
+func (aof *AppendableFile) write(bs []byte) (n int, err error) {
+	for n < len(bs) {
+		available := len(aof.writeBuffer) - aof.wbufUnwrittenOffset
+
+		if available == 0 {
+			if aof.retryableSync && !aof.autoSync {
```

Comment:
wdyt of making this as follows

```
	if aof.retryableSync{
		if !aof.autoSync {
			// Sync must be called to free buffer space
			return n, ErrBufferFull
		} else {
			err = aof.sync()
			if err != nil {
				return
			}	
		}
	} else {
		err = aof.flush()
		if err != nil {
			return
		}
	}

```

Also I am a bit unclear on what retryableSync does. I am assuming it forces flush and then sync? If so, wdyt about adding a small comment describing the purpose of it here?

Reply:
Ok my bad, just read the comment above in the struct definition of what retryableSync does. Pls ignore the comment line

Reply:
In general I prefer code readability, in the end compiler optimizations may be applied and in this type of changes performance shouldn't be affected. However, I made a slightly change there

Reply:
It was bit hard to grasp the logic with so many conditions, hence thought having a parental `if` would help. Subjective I guess, no blocker